File: Python/program/Dosen.py
""" Saya Apri Anggara Yudha NIM 2102585 mengerjakan soal Latihan4 dalam mata kuliah 
Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka saya 
tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. """

from SivitasAkademik import SivitasAkademik

class Dosen(SivitasAkademik):
    def __init__(self):
        self.__nip = ""
        self.__fakultas = ""
        self.__pend_terakhir = ""
        self.__keahlian = ""

    # Getter
    def getNip(self):
        return self.__nip
    def getFakultas(self):
        return self.__fakultas
    def getPendTerakhir(self):
        return self.__pend_terakhir
    def getKeahlian(self):
        return self.__keahlian

    # Setter
    def setNip(self, value):
        self.__nip = value
    def setFakultas(self, value):
        self.__fakultas = value
    def setPendTerakhir(self, value):
        self.__pend_terakhir = value
    def setKeahlian(self, value):
        self.__keahlian = value

File: CPP/program/Dosen.cpp
/* Saya Apri Anggara Yudha NIM 2102585 mengerjakan soal Latihan4 dalam mata kuliah 
Desain dan Pemrograman Berorientasi Objek untuk keberkahanNya maka saya 
tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. */

#include <string>
#include "SivitasAkademik.cpp"

using namespace std;

class Dosen : public SivitasAkademik {

private:
    string nip;
    string fakultas;
    string pend_terakhir;
    string keahlian;

public:
    Dosen () {
    }

    Dosen (string nip, string nama, string jenis_kelamin, string fakultas, string pend_terakhir, string keahlian)
    : SivitasAkademik(nama, jenis_kelamin) {
        this->nip = nip;
        this->fakultas = fakultas;
        this->pend_terakhir = pend_terakhir;
        this->keahlian = keahlian;
    }

    // Getter
    string getNip () {
        return nip;
    }
    string getFakultas () {
        return fakultas;
    }
    string getPendTerakhir () {
        return pend_terakhir;
    }
    string getKeahlian () {
        return keahlian;
    }

    // Setter
    void setNip (string nip) {
        this->nip = nip;
    }
    void setFakultas (string fakultas) {
        this->fakultas = fakultas;
    }
    void setPendTerakhir (string pend_terakhir) {
        this->pend_terakhir = pend_terakhir;
    }
    void setKeahlian (string keahlian) {
        this->keahlian = keahlian;
    }

};
